activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeTest.java

```diff
@@ -0,0 +1,325 @@
+
+    }
+
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void createProcessInstanceAndValidateHappyPath() {
+
+        ProcessRuntimeConfiguration configuration = processRuntime.configuration();
+        assertThat(configuration).isNotNull();
+        Page<ProcessDefinition> processDefinitionPage = processRuntime.processDefinitions(Pageable.of(0,
+                50));
+        assertThat(processDefinitionPage.getContent()).isNotNull();
+        assertThat(processDefinitionPage.getContent())
+                .extracting((ProcessDefinition pd) -> pd.getKey())
+                .contains("categorizeProcess");
+
+        ProcessInstance categorizeProcess = processRuntime.start(ProcessPayloadBuilder.start()
+                .withProcessDefinitionKey("categorizeProcess")
+                .withVariable("expectedKey",
+                        true)
+                .build());
+
+        assertThat(categorizeProcess).isNotNull();
+
+        assertThat(categorizeProcess.getStatus()).isEqualTo(ProcessInstance.ProcessInstanceStatus.COMPLETED);
+        assertThat(RuntimeTestConfiguraiton.processImageConnectorExecuted).isEqualTo(true);
+        assertThat(RuntimeTestConfiguraiton.tagImageConnectorExecuted).isEqualTo(true);
+        assertThat(RuntimeTestConfiguraiton.discardImageConnectorExecuted).isEqualTo(false);
+    }
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void createProcessInstanceAndValidateDiscardPath() {
+
+        ProcessRuntimeConfiguration configuration = processRuntime.configuration();
+        assertThat(configuration).isNotNull();
+        Page<ProcessDefinition> processDefinitionPage = processRuntime.processDefinitions(Pageable.of(0,
+                50));
+        assertThat(processDefinitionPage.getContent()).isNotNull();
+        assertThat(processDefinitionPage.getContent())
+                .extracting((ProcessDefinition pd) -> pd.getKey())
+                .contains("categorizeProcess");
+
+        ProcessInstance categorizeProcess = processRuntime.start(ProcessPayloadBuilder.start()
+                .withProcessDefinitionKey("categorizeProcess")
+                .withVariable("expectedKey",
+                        false)
+                .build());
+
+        assertThat(categorizeProcess).isNotNull();
+
+        assertThat(categorizeProcess.getStatus()).isEqualTo(ProcessInstance.ProcessInstanceStatus.COMPLETED);
+        assertThat(RuntimeTestConfiguraiton.processImageConnectorExecuted).isEqualTo(true);
+        assertThat(RuntimeTestConfiguraiton.tagImageConnectorExecuted).isEqualTo(false);
+        assertThat(RuntimeTestConfiguraiton.discardImageConnectorExecuted).isEqualTo(true);
+    }
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void getProcessInstances() {
```

Comment:
this test is a little bit long, it could take benefit from some method extractions.